src/SignalR/clients/ts/signalr/src/FetchHttpClient.ts

```diff
@@ -9,35 +9,36 @@ import { HttpClient, HttpRequest, HttpResponse } from "./HttpClient";
 import { ILogger, LogLevel } from "./ILogger";
 import { Platform } from "./Utils";
 
-let abortControllerType: { prototype: AbortController, new(): AbortController };
-let fetchType: (input: RequestInfo, init?: RequestInit) => Promise<Response>;
-let jar: tough.CookieJar;
-if (typeof fetch === "undefined") {
-    // In order to ignore the dynamic require in webpack builds we need to do this magic
-    // @ts-ignore: TS doesn't know about these names
-    const requireFunc = typeof __webpack_require__ === "function" ? __non_webpack_require__ : require;
-
-    // Cookies aren't automatically handled in Node so we need to add a CookieJar to preserve cookies across requests
-    jar = new (requireFunc("tough-cookie")).CookieJar();
-    fetchType = requireFunc("node-fetch");
-
-    // node-fetch doesn't have a nice API for getting and setting cookies
-    // fetch-cookie will wrap a fetch implementation with a default CookieJar or a provided one
-    fetchType = requireFunc("fetch-cookie")(fetchType, jar);
-
-    // Node needs EventListener methods on AbortController which our custom polyfill doesn't provide
-    abortControllerType = requireFunc("abort-controller");
-} else {
-    fetchType = fetch;
-    abortControllerType = AbortController;
-}
-
 export class FetchHttpClient extends HttpClient {
+    private readonly abortControllerType: { prototype: AbortController, new(): AbortController };
+    private readonly fetchType: (input: RequestInfo, init?: RequestInit) => Promise<Response>;
+    private readonly jar?: tough.CookieJar;
+
     private readonly logger: ILogger;
 
     public constructor(logger: ILogger) {
         super();
         this.logger = logger;
+
+        if (typeof fetch === "undefined") {
+            // In order to ignore the dynamic require in webpack builds we need to do this magic
+            // @ts-ignore: TS doesn't know about these names
+            const requireFunc = typeof __webpack_require__ === "function" ? __non_webpack_require__ : require;
+
+            // Cookies aren't automatically handled in Node so we need to add a CookieJar to preserve cookies across requests
+            this.jar = new (requireFunc("tough-cookie")).CookieJar();
+            this.fetchType = requireFunc("node-fetch");
+
+            // node-fetch doesn't have a nice API for getting and setting cookies
+            // fetch-cookie will wrap a fetch implementation with a default CookieJar or a provided one
+            this.fetchType = requireFunc("fetch-cookie")(this.fetchType, this.jar);
+
+            // Node needs EventListener methods on AbortController which our custom polyfill doesn't provide
+            this.abortControllerType = requireFunc("abort-controller");
+        } else {
+            this.fetchType = fetch.bind(self);
```

Comment:
Why call `bind`?

Reply:
Without calling the `bind`, it will cause error if you try to invoke `fetch`:
```
Network error: TypeError: Failed to execute 'fetch' on 'Window': Illegal invocation
```

It seems to be a bug coming from fetch, where a binding is set incorrectly. Calling `fetch.bind(self)` can help to resolve the problem.